src/dbnode/persist/fs/files.go

```diff
@@ -1220,3 +1336,75 @@ func snapshotIndexSegmentFilePathFromTimeAndIndex(
 	suffix := filesetIndexSegmentFileSuffixFromTime(t, segmentIndex, segmentFileType)
 	return filesetPathFromTimeAndIndex(prefix, t, snapshotIndex, suffix)
 }
+
+func snapshotMetadataFilePathFromIdentifier(prefix string, id SnapshotMetadataIdentifier) string {
+	return path.Join(
+		prefix,
+		snapshotDirName,
+		fmt.Sprintf(
+			"%s-%s-%d-%s%s",
+			snapshotFilePrefix, sanitizeUUID(id.UUID.String()), id.Index, metadataFileSuffix, fileSuffix))
+}
+
+func snapshotMetadataCheckpointFilePathFromIdentifier(prefix string, id SnapshotMetadataIdentifier) string {
+	return path.Join(
+		prefix,
+		snapshotDirName,
+		fmt.Sprintf(
+			"%s-%s-%d-%s-%s%s",
+			snapshotFilePrefix, sanitizeUUID(id.UUID.String()), id.Index, metadataFileSuffix, checkpointFileSuffix, fileSuffix))
+}
+
+// sanitizeUUID replaces all instances of separator ("-") with uuidSeparator ("_") in the provided
+// uuid string. If we did not do this, then when we tried to split up filenames based on the separator
+// every "piece" of the UUID would be included as a separate fragment from the name when instead we want
+// to treat the entire UUID as one "piece" of the name.
+func sanitizeUUID(uuid string) string {
```

Comment:
Ugh... this is pretty gross.  Instead can we just replace it with empty string?  That should be parseable by UUID libraries.